### PR TITLE
Add eslint-plugin-testing-library

### PR DESCRIPTION
**Pull request by ahuth**
### Summary:

This PR adds the eslint plugin for testing library.

### Test Plan:

- CI
- Added an error intentionally and ran linters locally

### CODE REVIEW

File: packages/components/src/Text/Text.spec.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ describe("<Text />", () => {
     textRef.current.focus();
 
     const textElement = screen.getByText("Some Text");
+    // eslint-disable-next-line testing-library/no-node-access
```
Comment (ahuth) [inline packages/components/src/Text/Text.spec.tsx:37]: The new eslint rules don't like directly accessing `document.activeElement`, it seems 🤔 

We could disable the rule. Or upgrade testing library and get access to `toHaveFocus`.

Comment (anniehu4) [reply packages/components/src/Text/Text.spec.tsx:37]: Or is the issue that we don't have `@testing-library/jest-dom` installed?

**Review by anniehu4 — APPROVED** [review]

File: packages/components/src/Text/Text.spec.tsx (hunk 1 above)
Comment (ahuth) [reply packages/components/src/Text/Text.spec.tsx:37]: That's probably it. I assumed we had that installed. But if not, that's definitely the problem.

**Review by dierat — APPROVED** [review]

**Comment by ahuth** [comment]
Thanks!

**Pull request merged**